server/CVCMIServer.cpp

```diff
@@ -615,9 +625,24 @@ int main(int argc, char** argv)
 		//and return non-zero status so client can detect error
 		throw;
 	}
+#ifdef VCMI_ANDROID
+	AndroidVMHelper envHelper;
+	envHelper.callStaticVoidMethod(AndroidVMHelper::NATIVE_METHODS_DEFAULT_CLASS, "killServer");
+#endif
 	delete VLC;
 	VLC = nullptr;
 	CResourceHandler::clear();
 
   return 0;
 }
+
+#ifdef VCMI_ANDROID
+
+void CVCMIServer::create()
+{
+	boost::this_thread::sleep(boost::posix_time::milliseconds(1000));

```

Comment:
About the whole design of interprocess communication then yeah, I can't use boost's interprocess. Currently client and server communicate through android Messanger (in java), which basically sends messages between processes. So basically client requests that the server should be started and when server finishes initializing, it sends the message that it's ready.